### DISCUSSION TITLE
times operator for tensor products in RankTwoTensor

### DISCUSSION THREAD

**Discussion by ambehnam** [idaholab/moose#21564] [Q&A Modules: Solid mechanics] [answered]
Hi there. I used the times operator to perform the tensor product shown in image below. Since 2 of the indices are repeated, the rank of the result (tau) should be a RankTwoTensor, but it outputs RankFourTensor. I was wondering if there is another way to perform this operation or convert the RankFour to RankTwo? I put part of the code and the equation below for more information.

  TauE1_3d_c(dv, nel, db, &intb, &vol); //db is a RankTwoTensor
  RankFourTensor C = _dpk1dF_e[_qp];
  usingTensorIndices(i_ , j_ , m_ , n_ );
  const RankFourTensor tau = db.times<i_,j_,m_,i_,n_,j_>(C); 

![tau_eqn](https://user-images.githubusercontent.com/40694609/178556986-328c305f-7613-4026-a1ee-8835a161fa2e.png)

Thanks,
Amir

**Comment by GiudGiud** [comment, marked as answer]
Hello

Not sure, please check the doxygen for this info
https://mooseframework.inl.gov/docs/doxygen/moose/classRankFourTensorTempl.html

Guillaume

**Reply by hugary1995** [reply]
This operator is indeed missing. I plan to work on it. Sorry for the delay.
